lib/ex_aws/sns.ex

```diff
@@ -161,22 +161,77 @@ defmodule ExAws.SNS do
   @doc "List Subscriptions"
   @spec list_subscriptions() :: ExAws.Operation.Query.t
   def list_subscriptions() do
-    request("ListSubscriptions", %{})
+    request(:list_subscriptions, %{})
   end
 
   @spec list_subscriptions(next_token :: binary) :: ExAws.Operation.Query.t
   def list_subscriptions(next_token) do
-    request("ListSubscriptions", %{"NextToken" => next_token})
+    request(:list_subscriptions, %{"NextToken" => next_token})
   end
 
+  ## Endpoints
+  ######################
+
+  @type endpoint_arn :: binary
+
+  @type endpoint_attributes :: [
+    {:token, binary}
+    | {:enabled, boolean}
+    | {:custom_user_data, binary}
+  ]
+
+  @doc "Get endpoint attributes"
+  @spec get_endpoint_attributes(endpoint_arn :: endpoint_arn) :: ExAws.Operation.Query.t
+  def get_endpoint_attributes(endpoint_arn) do
+    request(:get_endpoint_attributes, %{"EndpointArn" => endpoint_arn})
+  end
+
+  @doc "Set endpoint attributes"
+  @spec set_endpoint_attributes(endpoint_arn :: endpoint_arn, attributes :: endpoint_attributes) :: ExAws.Operation.Query.t
+  def set_endpoint_attributes(endpoint_arn, attributes) do
+    params =
+      attributes
+      |> build_attrs
+
+      |> IO.inspect
+
+    request(:set_endpoint_attributes, Map.put(params, "EndpointArn", endpoint_arn))
+  end
+
+
   ## Request
   ######################
 
   defp request(action, params) do
+    action_string = action |> Atom.to_string |> Macro.camelize
+
     %ExAws.Operation.Query{
       path: "/",
-      params: params |> Map.put("Action", action),
-      service: :sns
+      params: params |> Map.put("Action", action_string),
+      service: :sns,
+      action: action,
+      parser: &ExAws.SNS.Parsers.parse/2
     }
   end
+
+  defp build_attrs(attrs) do
```

Comment:
These private methods are mostly stolen from SQS, but they are not the exact same, because the format is slightly different for SNS (thanks Amazon). Not sure how you feel about duplicating these here.